common/src/auth/ServiceConnectionStringParser.cs

```diff
@@ -0,0 +1,228 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.DigitalTwin.Model.Service;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Azure.Devices.Common.Authorization
+{
+    /// <summary>
+    /// The Service Connection String Parser class
+    /// </summary>
+    public class ServiceConnectionStringParser
+    {
+        protected char ValuePairDelimiter = ';';
+        protected char ValuePairSeparator = '=';
+        protected static string HostNameSeparator = ".";
+
+        protected string HostNamePropertyName = nameof(HostName);
+        protected string SharedAccessKeyNamePropertyName = nameof(SharedAccessKeyName);
+        protected string SharedAccessKeyPropertyName = nameof(SharedAccessKey);
+        protected string SharedAccessSignaturePropertyName = nameof(SharedAccessSignatureString);
+
+        protected static TimeSpan regexTimeoutMilliseconds = TimeSpan.FromMilliseconds(500);
+        protected Regex HostNameRegex = new Regex(@"[a-zA-Z0-9_\-\.]+$", RegexOptions.Compiled | RegexOptions.IgnoreCase, regexTimeoutMilliseconds);
+        protected Regex SharedAccessKeyNameRegex = new Regex(@"^[a-zA-Z0-9_\-@\.]+$", RegexOptions.Compiled | RegexOptions.IgnoreCase, regexTimeoutMilliseconds);
+        protected Regex SharedAccessKeyRegex = new Regex(@"^.+$", RegexOptions.Compiled | RegexOptions.IgnoreCase, regexTimeoutMilliseconds);
+        protected Regex SharedAccessSignatureRegex = new Regex(@"^.+$", RegexOptions.Compiled | RegexOptions.IgnoreCase, regexTimeoutMilliseconds);
+
+        private string _hostName;
+        private string _serviceName;
+
+        protected ServiceConnectionStringParser()
+        {
+        }
+
+        /// <summary>
+        /// Factory for new Connection String object.
+        /// </summary>
+        /// <remarks>
+        /// The connection string contains a set of information that uniquely identify an IoT Service.
+        ///
+        /// A valid connection string shall be in the following format:
+        /// <code>
+        /// HostName=[ServiceName];SharedAccessKeyName=[keyName];SharedAccessKey=[Key]
```

Comment:
HostName is the ServiceName? Not the service endpoint?

Reply:
I changed it according to model repository document